packages/task/src/node/task-server.ts

```diff
@@ -0,0 +1,258 @@
+            // new process, so e.g. we can re-use the system path
+            processOptions.options = {
+                cwd: cwd,
+                env: process.env
+            };
+
+            // When we create a process to execute a command, it's difficult to know if it failed
+            // because the executable or script was not found, or if it was found, ran, and exited
+            // unsuccessfully. So here we look to see if it seems we can find a file of that name that
+            // is likely to be the one we want, before attemting to execute it.
+            this.findCommand(command, cwd)
+                .then(cmd => {
+                    try {
+                        // use terminal or raw process
+                        let task: Task;
+                        let proc: TerminalProcess | RawProcess;
+
+                        if (options.processType === 'terminal') {
+                            this.logger.info('Task: creating underlying terminal process');
+                            proc = this.terminalProcessFactory(<TerminalProcessOptions>processOptions);
+                        } else {
+                            this.logger.info('Task: creating underlying raw process');
+                            proc = this.rawProcessFactory(<RawProcessOptions>processOptions);
+                        }
+
+                        task = this.taskFactory(<TaskProcessOptions>
+                            {
+                                label: options.label,
+                                command: cmd,
+                                process: proc,
+                                processType: options.processType,
+                                context: ctx
+                            });
+
+                        const toDispose = new DisposableCollection();
+
+                        toDispose.push(
+                            // when underlying process exits, notify tasks listeners
+                            proc.onExit(event => {
+
+                                this.fireTaskExitedEvent({
+                                    'taskId': task.id,
+                                    'code': event.code,
+                                    'signal': event.signal,
+                                    'ctx': ctx === undefined ? '' : ctx
+                                });
+                            })
+                        );
+
+                        this.taskToDispose.set(task.id, toDispose);
+
+                        const newTask: TaskInfo = {
+                            taskId: task.id,
+                            osProcessId: proc.pid,
+                            terminalId: (options.processType === 'terminal') ? proc.id : undefined,
+                            processId: (options.processType === 'raw') ? proc.id : undefined,
+                            command: cmd,
+                            label: options.label,
+                            ctx: ctx === undefined ? '' : ctx
```

Comment:
Could we extract the translation to a method and use it also from here https://github.com/theia-ide/theia/pull/690/files#diff-fccfd5a001011db19d656ecd3d66d03dR67

Reply:
good idea - done